### PR TITLE
Feature/shift walker plugin

### PR DESCRIPTION
**Pull request by plinko60**
Feature requested in https://github.com/runelite/runelite/issues/1412 .

This plugin was built based off the work in the MenuEntrySwapperPlugin . This works very similar to how that plugin works, if this is more suited to extend the MenuEntrySwapperPlugin that can be done, however more extensive changes are needed to accommodate multiple entries being tied to the shift hotkey (MenuEntry will currently only handle 1).

### CODE REVIEW

**Comment by plinko60** [comment]
Current features:
**Walk Under Everything** - Enable this option when you do not want to interact with anything while Shift is pressed. If Walk Here is an option it will be the action taken.

**Walk Under Box Traps** - Press "Shift" to be able to walk under instead of picking up a Box Trap.

**Walk Under Attack Options** - Press "Shift" to be able to walk instead of attacking. Make sure Left Click Attack is on.

**Comment by plinko60** [comment]
So there was conversation on discord about this being potentially too strong for PvP situations. (Such as barrage and walk under).

If requested, I can look into making this not work for Pvp. The change would be to disable this walk under when you have an attack option on the player.

File: runelite-client/src/main/java/net/runelite/client/plugins/shiftwalker/ShiftWalkerPlugin.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	 * Finds the index of the menu that contains the verbiage we are looking for.
+	 * @param pMenuEntries The list of {@link MenuEntry}s.
+	 * @param pMenuEntryToSearchFor The Option in the menu to search for.
+	 * @return The index location or null if it was not found.
+	 */
+	private Integer searchIndex(MenuEntry[] pMenuEntries, String pMenuEntryToSearchFor)
+	{
+		Integer indexLocation = 0;
+
+		for (MenuEntry menuEntry : pMenuEntries)
```
Comment (plinko60) [inline runelite-client/src/main/java/net/runelite/client/plugins/shiftwalker/ShiftWalkerPlugin.java]: This needs to be a search in the reverse order than this. Searching like this only does the top level entry. But multiple entries in the menu won't get touched, meaning this won't work then.

**Comment by deathbeam** [comment]
See reply in #1412

**Pull request closed without merging**